lib/parsers/iso_base_media_file_format/decoder.rb

```diff
@@ -1,30 +1,42 @@
-# This class provides generic methods for parsing file formats based on QuickTime-style "atoms", such as those seen in
+# This class provides generic methods for parsing file formats based on QuickTime-style "boxes", such as those seen in
 # the ISO base media file format (ISO/IEC 14496-12), a.k.a MPEG-4, and those that extend it (MP4, CR3, HEIF, etc.).
 #
-# For more information on atoms, see https://developer.apple.com/library/archive/documentation/QuickTime/QTFF/QTFFChap1/qtff1.html
+# For more information on boxes, see https://developer.apple.com/library/archive/documentation/QuickTime/QTFF/QTFFChap1/qtff1.html
 # or https://b.goeswhere.com/ISO_IEC_14496-12_2015.pdf.
 #
 # TODO: The vast majority of the methods have been commented out here. This decision was taken to expedite the release
 #   of support for the CR3 format, such that it was not blocked by the undertaking of testing this class in its
 #   entirety. We should migrate existing formats that are based on the ISO base media file format and reintroduce these
 #   methods with tests down-the-line.
 
+require 'matrix'
+
 module FormatParser
   module ISOBaseMediaFileFormat
     class Decoder
       include FormatParser::IOUtils
 
-      class Atom < Struct.new(:type, :position, :size, :fields, :children)
+      class Box < Struct.new(:type, :position, :size, :fields, :children)
         def initialize(type, position, size, fields = nil, children = nil)
           super
           self.fields ||= {}
           self.children ||= []
         end
 
+        def [](index)
+          if index.is_a?(Symbol)
```

Comment:
I find this a bit confusing. Wouldn't expect that addressing this class by symbol or string would look in different places for that particular element 🤔

Reply:
I thought the pros out-weighed the cons in favour of this approach over separate, dedicated methods - I think it gives more clarity when trying to access deeply nested boxes, e.g. `trak_box['mdia']['minf']['stbl']['stsd']` vs. `trak_box.get_child('mdia').get_child('minf).get_child('stbl').get_child('stsd')` vs `trak_box.get_descendent(['mdia', 'minf', 'stbl', 'stsd']`.

I don't feel especially strongly about it though, so if you do then I'm happy to change it up. ✌🏻

Reply:
I'm fine either way. Just pointed out that it looks confusing at first sight; I trust you had a better overview of the usage of this 🙌